agent/xds/listeners.go

```diff
@@ -1022,6 +1023,19 @@ func (s *ResourceGenerator) injectConnectTLSForPublicListener(cfgSnap *proxycfg.
 	return nil
 }
 
+func getAlpnProtocols(protocol string) []string {
+	var alpnProtocols []string
+
+	switch protocol {
+	case "grpc", "http2":
```

Comment:
I believe `grpc` and `http2` cases behave differently. `http2` should only return `h2`.

Reply:
Thanks! I've updated the function to return `h2` only for `http2`.